URL action

sbliven opened the pull request:
Initial URLAction implementation.

The test configuration simply calls /health and logs the result.

sbliven committed:
Initial urlaction implementation

The test configuration simply calls /health and logs the result.

nitrosx commented on src/config/configuration.ts:
```diff
@@ -46,7 +47,9 @@ const configuration = () => {
   registerDefaultActions();
-  const job_configs: Promise<JobConfig[]> = loadJobConfig("src/jobs/config/jobConfig.createExample.json");
+  const job_configs: Promise<JobConfig[]> = loadJobConfig(
+    "src/jobs/config/jobConfig.createExample.json",
```
We should brainstorm about how to remove the hardcoding and make it more site configurable.

sbliven replied:
Yes, there's a lot of little things like that that will be quick to implement but need to be handled before release.

sbliven replied:
#1114

nitrosx reviewed (approved):
Just left a comment as reminder to discuss the issue when the functionalities are stable

sbliven committed:
Add 'jsonify' Handlebars helper

sbliven committed:
Accept template for URLAction config

This is a restrictive implementation; for more on security, see
https://github.com/SciCatProject/scicat-backend-next/issues/1113

sbliven commented on src/jobs/actions/urlaction.ts:
```diff
@@ -0,0 +1,91 @@
+    createdBy: true,
+    jobParams: false,
+  },
+  allowProtoPropertiesByDefault: false, // limit accessible fields for security
```
I went with a more restrictive setting for now while trying to understand the security. See #1113 for details.

This approach is harder to maintain since the templatable properties need to be kept up to date with the JobClass schema.

sbliven commented on src/jobs/actions/urlaction.ts:
```diff
@@ -0,0 +1,91 @@
+  private urlTemplate: Handlebars.TemplateDelegate<JobClass>;
+  private method = "GET";
+  private headers: Record<string, string> = {};
+  private body: Record<string, any> | null = null;
```
Body probably also needs to accept templates, but how to deal with it in JSON input? Maybe only apply templating to string values?

sbliven commented on src/jobs/actions/urlaction.ts:
```diff
@@ -0,0 +1,91 @@
+      );
+    }
+
+    // TODO do something with the response?
```
Should the response body be used to change the jobstatus?

sbliven commented:
This is done for now. I'm going to merge into release-jobs and make any changes there.

sbliven committed:
Merge  branch 'release-jobs' into urlaction
